src/test/java/org/galatea/starter/entrypoint/IexRestControllerTest.java

```diff
@@ -75,10 +76,80 @@ public void testGetLastTradedPriceEmpty() throws Exception {
-            .get("/iex/lastTradedPrice?symbols=")
+            .get("/iex/lastTradedPrice?token=abc&symbols=")
             .accept(MediaType.APPLICATION_JSON_VALUE))
         .andExpect(status().isOk())
         .andExpect(jsonPath("$", is(Collections.emptyList())))
         .andReturn();
   }
+
+  @Test
+  public void testGetHistoricalPriceWithRange() throws Exception {
+
+    MvcResult result = this.mvc.perform(
+            org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+                .get("/iex/historicalPrice?symbol=AAPL&range=6m&token=abc")
+                .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].symbol", is("AAPL")))
+        .andExpect(jsonPath("$[0].close").value(new BigDecimal("180.33")))
+        .andExpect(jsonPath("$[1].symbol", is("AAPL")))
+        .andExpect(jsonPath("$[1].high").value(new BigDecimal("181.33")))
+        .andExpect(jsonPath("$[2].date", is("2021-12-29")))
+        .andReturn();
+  }
+
+  @Test
+  public void testGetHistoricalPriceWithDate() throws Exception {
+
+    MvcResult result = this.mvc.perform(
+            org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+                .get("/iex/historicalPrice?symbol=AAPL&date=20220104&token=abc")
+                .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].symbol", is("AAPL")))
+        .andExpect(jsonPath("$[0].close").value(new BigDecimal("182.53")))
+        .andExpect(jsonPath("$[0].date", is("2022-01-04")))
+        .andExpect(jsonPath("$[1].symbol", is("AAPL")))
+        .andExpect(jsonPath("$[1].high").value(new BigDecimal("182.66")))
+        .andExpect(jsonPath("$[2].date", is("2022-01-04")))
+        .andReturn();
+  }
+
+  @Test
+  public void testGetHistoricalPriceBase() throws Exception {
+
+    MvcResult result = this.mvc.perform(
+            org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+                .get("/iex/historicalPrice?symbol=AAPL&token=abc")
+                .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].symbol", is("AAPL")))
+        .andExpect(jsonPath("$[0].close").value(new BigDecimal("140.36")))
+        .andExpect(jsonPath("$[1].symbol", is("AAPL")))
+        .andExpect(jsonPath("$[1].high").value(new BigDecimal("141.785")))
+        .andReturn();
+  }
+
+  @Test
+  public void testGetHistoricalPriceNoSymbol() throws Exception {
+
+    MvcResult result = this.mvc.perform(
+            org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+                .get("/iex/historicalPrice?token=abc")
+                .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isBadRequest())
+        .andReturn();
+  }
+
+  @Test(expected = NestedServletException.class)
+  public void testGetHistoricalPriceInvalidSymbol() throws Exception {
```

Comment:
Add tests for when IEX responds incorrectly, e.g. with an internal server error or not found